controller.go

```diff
@@ -266,7 +270,29 @@ func updateAlarmDefinition(id string, r Resource) error {
 	definitionRequest := convertToADRequest(r.Spec.AlarmDefinitionElement)
 	result, err := monascaclient.PatchAlarmDefinition(id, definitionRequest)
 	if err != nil {
-		return err
+		if !strings.HasPrefix(err.Error(), "Error: 422") {
+			return err
+		}
+		// Some updates are not allowed, so try deleting and recreating instead
+		// NOTE: this will remove the alarms under this definition and thus remove
+		// any alarm history as well.
+		log.Printf("Failed to update alarm %s, attempting delete and recreate", id)
```

Comment:
do we have the name the id is going to be confusing to read in the log statement